### DISCUSSION TITLE
Is this the expected behavior of dict() exclude_unset on default values?

### DISCUSSION THREAD

**Discussion by makilegaspi** [pydantic/pydantic#3136] [Question]
Seems like the default values are ignored when doing a `.dict(exclude_unset=True)`. It's a bit confusing and I expected they're still included. This happens whether I explicitly add an `exclude_defaults=False` or not.

```python
from pydantic import BaseModel


class Foo(BaseModel):
    name: str
    other: str = "Dummy"


foo = Foo(name="Joe")

print(foo)
# Output: name='Joe' other='Dummy'

print(foo.dict(exclude_unset=True))
# Expected Output: {'name': 'Joe', 'other': 'Dummy'}
# Actual Output: {'name': 'Joe'}

```

**Comment by alxmoroz** [comment]
May be its bug?

**Comment by maxwelllevin** [comment]
I also encountered this same issue and found that `exclude_none=True` does the trick for what I needed.

```python
from pydantic import BaseModel


class Foo(BaseModel):
    name: str
    other: str = "Dummy"


foo = Foo(name="Joe")

print(foo)
# Output: name='Joe' other='Dummy'

print(foo.dict(exclude_none=True))
# Expected Output: {'name': 'Joe', 'other': 'Dummy'}
# Actual Output: {'name': 'Joe', 'other': 'Dummy'}
```

**Comment by alxmoroz** [comment]
I found a way in my case to use private fields. They don't get into field generation in SQLAlchemy models
Another working approach is to use different models for creating / updating in the database and for reading. But too much code is required (more code, more bugs, etc.).

**Comment by alxmoroz** [comment]
@makilegaspi @maxwellevin 
Please vote for issue to still open or close: #3851

**Reply by makilegaspi** [reply]
I think we can close the issue. I just read your comment and found that it was indeed stated in the documents that those that are not **_explicitly_** set are excluded.
